src/tests/rules/prefer-repository-shorthand.test.ts

```diff
@@ -0,0 +1,107 @@
+import rule from "../../rules/prefer-repository-shorthand.js";
+import { ruleTester } from "./ruleTester.js";
+
+ruleTester.run("prefer-repository-shorthand", rule, {
+	invalid: [
+		{
+			code: `{
+	"repository": {
+		"type": "git",
+		"url": "https://github.com/JoshuaKGoldberg/create-typescript-app"
+	}
+}`,
+			errors: [
+				{
+					message:
+						"Use shorthand repository URL for GitHub repository",
+					type: "JSONObjectExpression",
+				},
+			],
+			filename: "package.json",
+			output: `{
+	"repository": "JoshuaKGoldberg/create-typescript-app"
+}`,
+		},
+		{
+			code: `{
+	"repository": {
+	  "type": "git",
+	  "url": "git+https://github.com/vuejs/core.git"
+	}
+}`,
+			errors: [
+				{
+					message:
+						"Use shorthand repository URL for GitHub repository",
+					type: "JSONObjectExpression",
+				},
+			],
+			filename: "package.json",
+			output: `{
+	"repository": "vuejs/core"
+}`,
+		},
+		{
+			code: `{
+	"repository": {
+		"type": "git",
+		"url": "git+ssh://git@github.com/yudai-nkt/eslint-plugin-uvu.git"
+	}
+}`,
+			errors: [
+				{
+					message:
+						"Use shorthand repository URL for GitHub repository",
+					type: "JSONObjectExpression",
+				},
+			],
+			filename: "package.json",
+			output: `{
+	"repository": "yudai-nkt/eslint-plugin-uvu"
+}`,
+		},
+		{
+			code: `{
+	"repository": "https://github.com/eslint/eslint"
+}`,
+			errors: [
+				{
+					message:
+						"Use shorthand repository URL for GitHub repository",
+					type: "JSONLiteral",
+				},
+			],
+			filename: "package.json",
+			output: `{
+	"repository": "eslint/eslint"
+}`,
+		},
+	],
+	valid: [
```

Comment:
[Testing] Thinking of a couple more `valid` cases...

* A weird repository like `a/b/c`
* A custom nested property like `{ "outer": { "repository": "https://github.com/a/b" } }`